[libzmq 4.2.5] Android ndk17 arm64-v8a compile error v1_decoder.cpp:113:28

libzmq 4.2.5
Android
ndk17
arm64-v8a

compile error:
  CXX      src/src_libzmq_la-v2_encoder.lo
../src/v1_decoder.cpp:113:28: error: comparison 'unsigned long' > 18446744073709551615 is always false [-Werror,-Wtautological-constant-compare]
    if (payload_length - 1 > std::numeric_limits<size_t>::max ()) {
        ~~~~~~~~~~~~~~~~~~ ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -35,6 +35,18 @@ if(COMPILER_SUPPORTS_C11)
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=gnu11")
 endif()
 
+if(NOT MSVC)
+    # clang 6 has a warning that does not make sense on multi-platform code
+    CHECK_CXX_COMPILER_FLAG("-Wno-tautological-constant-compare" CXX_HAS_TAUT_WARNING)
+    if(CXX_HAS_TAUT_WARNING)
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-tautological-constant-compare")
+    endif()
+    CHECK_C_COMPILER_FLAG("-Wno-tautological-constant-compare" CC_HAS_TAUT_WARNING)
+    if(CC_HAS_TAUT_WARNING)
+        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-tautological-constant-compare")
+    endif()
+endif()
+
 # Will be used to add flags to pkg-config useful when apps want to statically link
 set (pkg_config_libs_private "")
 

--- a/configure.ac
+++ b/configure.ac
@@ -812,6 +812,12 @@ AM_CONDITIONAL([WITH_CLANG_FORMAT], [$WITH_CLANG_FORMAT])
 # unittests will not build without the static libzmq.a
 AM_CONDITIONAL(ENABLE_STATIC, test "x$enable_static" = "xyes")
 
+# clang 6 has a warning that does not make sense on multi-platform code
+AX_CHECK_COMPILE_FLAG([-Wno-tautological-constant-compare],
+    [CFLAGS+=" -Wno-tautological-constant-compare" CXXFLAGS+=" -Wno-tautological-constant-compare"],
+    [],
+    [-Werror])
+
 # Subst LIBZMQ_EXTRA_CFLAGS & CXXFLAGS & LDFLAGS
 AC_SUBST(LIBZMQ_EXTRA_CFLAGS)
 AC_SUBST(LIBZMQ_EXTRA_CXXFLAGS)

--- a/src/tipc_connecter.cpp
+++ b/src/tipc_connecter.cpp
@@ -255,7 +255,7 @@ zmq::fd_t zmq::tipc_connecter_t::connect ()
     //  Following code should handle both Berkeley-derived socket
     //  implementations and Solaris.
     int err = 0;
-#if ZMQ_HAVE_VXWORKS
+#ifdef ZMQ_HAVE_VXWORKS
     int len = sizeof (err);
 #else
     socklen_t len = sizeof (err);

